### PR TITLE
HWY-205: Some cleanups related to LNC.

### PR DESCRIPTION
**Pull request by afck**
This will make the diff smaller when we merge LNC-compliant unit creation.
* Add a `Panorama::iter_faulty` to deduplicate some code.
* `citable_panorama` will be replaced with something simpler, so I remove it here.
* Rename a few identifiers that were overlooked or renamed by mistake.

https://casperlabs.atlassian.net/browse/HWY-205

### CODE REVIEW

File: node/src/components/consensus/highway_core/state.rs [hunk 1]
```diff
@@ -672,29 +609,30 @@ impl<C: Context> State<C> {
...
+        let unit = self.unit(&uhash);
+        let creator = unit.creator;
+        let new_obs = match (self.panorama.get(creator), unit.panorama.get(creator)) {
             (Observation::Faulty, _) => Observation::Faulty,
-            (obs0, obs1) if obs0 == obs1 => Observation::Correct(wunit.hash()),
+            (obs0, obs1) if obs0 == obs1 => Observation::Correct(uhash),
             (Observation::None, _) => panic!("missing creator's previous unit"),
             (Observation::Correct(hash0), _) => {
                 // If we have all dependencies of wunit and still see the sender as correct, the
                 // predecessor of wunit must be a predecessor of hash0. So we already have a
```
Comment (fizyk20) [inline node/src/components/consensus/highway_core/state.rs:626]:
```suggestion
                // If we have all dependencies of unit and still see the sender as correct, the
                // predecessor of unit must be a predecessor of hash0. So we already have a
```

**Review by fizyk20 — APPROVED** [review]
Looking good :+1:

File: node/src/components/consensus/highway_core/state.rs (hunk 1 above)
Comment (afck) [reply node/src/components/consensus/highway_core/state.rs]: Good catch, thanks! :+1:

**Comment by afck** [comment]
bors merge

**Pull request merged**